predicators/envs/narrow_passage.py

```diff
@@ -0,0 +1,490 @@
+        y_mid = (self.y_ub - self.y_lb) / 2 + self.y_lb
+        margin = self.wall_thickness_half + self.init_pos_margin
+        door_width = (self.robot_radius + self.door_width_padding) * 2
+        passage_width = (self.robot_radius + self.passage_width_padding) * 2
+
+        tasks: List[Task] = []
+        while len(tasks) < num:
+            state = utils.create_state_from_dict({
+                self._robot: {
+                    "x":
+                    rng.uniform(self.x_lb, self.x_ub),
+                    "y":
+                    rng.uniform(y_mid + margin + self.robot_radius, self.y_ub),
+                },
+                self._target: {
+                    "x":
+                    rng.uniform(self.x_lb, self.x_ub),
+                    "y":
+                    rng.uniform(self.y_lb,
+                                y_mid - margin - self.target_radius),
+                },
+                # wall and door positions are fixed, defined by class variables
+                self._walls[0]: {
+                    "x": self.x_lb - self.robot_radius,
+                    "width": self.door_x_pos + self.robot_radius,
+                },
+                self._walls[1]: {
+                    "x": self.door_x_pos + door_width,
+                    "width": self.passage_x_pos - self.door_x_pos - door_width,
+                },
+                self._walls[2]: {
+                    "x":
+                    self.passage_x_pos + passage_width,
+                    "width":
+                    self.x_ub - self.passage_x_pos - passage_width +
+                    self.robot_radius,
+                },
+                self._door: {
+                    "x": self.door_x_pos,
+                    "open": 0,  # door starts closed
+                },
+                self._door_sensor: {
+                    "x": self.door_x_pos + door_width / 2,
+                },
+            })
+            # Make sure goal is not satisfied.
+            assert not goal_atom.holds(
+                state
+            ), "Error: goal is already satisfied in this state initialization"
+            tasks.append(Task(state, goal))
+        return tasks
+
+    @staticmethod
+    def _MoveToTarget_policy(state: State, memory: Dict,
+                             objects: Sequence[Object],
+                             params: Array) -> Action:
+        del state, objects, params  # unused
+        assert memory["action_plan"], "Motion plan did not reach its goal"
+        return memory["action_plan"].pop(0)
+
+    def _MoveToTarget_initiable(self, state: State, memory: Dict,
+                                objects: Sequence[Object],
+                                params: Array) -> bool:
+        robot, target = objects
+        # Set up the target input for the motion planner.
+        target_x = state.get(target, "x")
+        target_y = state.get(target, "y")
+        result = self._run_birrt(state, memory, params, robot,
```

Comment:
can we change this variable name to `success` or something similar? (here and elsewhere)